keras_nlp/src/models/segment_anything/segment_anything.py

```diff
@@ -0,0 +1,327 @@
+# Copyright 2024 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import keras
+import numpy as np
+from keras import ops
+
+from keras_nlp.src.api_export import keras_nlp_export
+from keras_nlp.src.models.segment_anything.sam_backbone import SAMBackbone
+from keras_nlp.src.models.segment_anything.sam_mask_decoder import (
+    SAMMaskDecoder,
+)
+from keras_nlp.src.models.segment_anything.sam_prompt_encoder import (
+    SAMPromptEncoder,
+)
+from keras_nlp.src.models.segment_anything.sam_transformer import (
+    TwoWayTransformer,
+)
+from keras_nlp.src.models.task import Task
+
+
+@keras_nlp_export("keras_nlp.models.SegmentAnythingModel")
+class SegmentAnythingModel(Task):
```

Comment:
Don't directly subclass task here. Add an `ImageSegmenter` base class (which doesn't need to have much if anything on it yet), and subclass that here. https://github.com/keras-team/keras-nlp/pull/1820/ shows this.

Name this to agree with other symbols name and include the task, so `SAMImageSegmenter`.

Reply:
done!